### PR TITLE
Documentation for Tessera AWS Secrets Manager implementation

### PR DESCRIPTION
**Pull request by Emi14**

### CODE REVIEW

**Review by Krish1979 — DISMISSED** [review]

File: docs/Privacy/Tessera/Configuration/Keys.md [hunk 1]
```diff
@@ -19,6 +19,9 @@ You can use existing private/public key pairs as well as use Tessera to generate
         "tlsKeyStorePath": "Path",
         "tlsTrustStorePath": "Path" 
     },
+    "awsKeyVaultConfig":{
```
Comment (chris-j-h) [inline docs/Privacy/Tessera/Configuration/Keys.md]:
```suggestion
    "keyVaultConfig": {
        "keyVaultConfigType": "AWS",
        "properties": {
            "endpoint": "Url"
        }
    },
```

File: docs/Privacy/Tessera/Configuration/Keys.md [hunk 2]
```diff
@@ -19,6 +19,9 @@ You can use existing private/public key pairs as well as use Tessera to generate
...
+          "endpoint": "Url"
```
Comment (chris-j-h) [inline docs/Privacy/Tessera/Configuration/Keys.md]:
```suggestion
```

File: docs/Privacy/Tessera/Configuration/Keys.md [hunk 3]
```diff
@@ -219,6 +222,30 @@ Tessera requires TLS certificates and keys to be stored in `.jks` Java keystore
 !!! info
     If using a Hashicorp Vault additional environment variables must be set and a version 2 K/V secret engine must be enabled.  For more information see [Setting up a Hashicorp Vault](../../Tessera%20Services/Keys/Setting%20up%20a%20Hashicorp%20Vault).
 
+### AWS Secrets Manager key pairs
+The keys in the pair are stored as secrets in the _AWS Secrets Manager_.  This requires providing the secret IDs for both keys.  The endpoint is optional as the _AWS SDK_ can fallback to its inbuilt property retrieval chain (e.g. using the environment variable `AWS_REGION` or `~/.aws/config` file - see [the AWS docs](https://docs.aws.amazon.com/sdk-for-java/v2/developer-guide/credentials.html) for similar behaviour explained in the context of credentials):
+```json
+"keys": {
+        "awsKeyVaultConfig":{
```
Comment (chris-j-h) [inline docs/Privacy/Tessera/Configuration/Keys.md]:
```suggestion
        "keyVaultConfig": {
            "keyVaultConfigType": "AWS",
            "properties": {
                "endpoint": "https://secretsmanager.us-west-2.amazonaws.com"
            }
        },
```

File: docs/Privacy/Tessera/Configuration/Keys.md [hunk 4]
```diff
@@ -219,6 +222,30 @@ Tessera requires TLS certificates and keys to be stored in `.jks` Java keystore
...
+            "endpoint": "https://secretsmanager.us-west-2.amazonaws.com"
```
Comment (chris-j-h) [inline docs/Privacy/Tessera/Configuration/Keys.md]:
```suggestion
```

File: docs/Privacy/Tessera/Configuration/Keys.md [hunk 5]
```diff
@@ -219,6 +222,30 @@ Tessera requires TLS certificates and keys to be stored in `.jks` Java keystore
...
+        },
```
Comment (chris-j-h) [inline docs/Privacy/Tessera/Configuration/Keys.md]:
```suggestion
```

File: docs/Privacy/Tessera/Configuration/Keys.md [hunk 6]
```diff
@@ -19,6 +19,9 @@ You can use existing private/public key pairs as well as use Tessera to generate
         "tlsKeyStorePath": "Path",
         "tlsTrustStorePath": "Path" 
     },
+    "awsKeyVaultConfig":{
+          "endpoint": "Url"
+      }
```
Comment (chris-j-h) [inline docs/Privacy/Tessera/Configuration/Keys.md]:
```suggestion
```

**Review by chris-j-h — CHANGES_REQUESTED** [review]
Please make the suggested changes so that the docs reflect our updated config

**Comment by Emi14** [comment]
@chris-j-h could you please take another look?

**Review by chris-j-h — DISMISSED** [review]
looks good

**Review by Krish1979 — DISMISSED** [review]

**Comment by jpmsam** [comment]
Thank you @Emi14 and Ionut. We'll merge it after the Tessera release.

**Review by Krish1979 — APPROVED** [review]

**Pull request merged**